tree_traversal.go

```diff
@@ -53,6 +53,36 @@ func traversalContext4_16_256(childZeroIdx int) *traverse4_16_256 {
 	}
 }
 
+// traverseDesc4_16_256 is a context for traversing nodes with 4, 16, or 256 children in descending order.
+type traverseDesc4_16_256 struct {
+	numChildren     int
+	childZeroActive bool
+	childCurIdx     int
+}
+
+// nextChildIdx returns the index of the next child node to traverse.
+func (tc *traverseDesc4_16_256) nextChildIdx() (int, bool) {
+	if tc.childZeroActive {
```

Comment:
TBH, I'm not sure about this part, I've given the spec a read but I've discovered ART fairly recently so I might be missing something here.